transport_plugins/bled112/iotile_transport_bled112/virtual_bled112.py

```diff
@@ -55,9 +56,15 @@ class BLED112VirtualInterface(VirtualIOTileInterface):
     HighspeedHandle = 21
     TracingHandle = 23
 
+    AdvertisingPacketVer = 1
+    json_args = None
+
     def __init__(self, args):
         super(BLED112VirtualInterface, self).__init__()
 
+        if args:
+            json_args = args
+
```

Comment:
Actually, I see that you are deferring the check on what is contained in `args` until you use it.  A better way is to capture the values you need from args with their defaults if not present in __init__, validate them to be valid and then you know their values are valid in the rest of the methods:

```python
advertising_version = args.get('advertising_version', 1)
if advertising_version not in (1, 2):
    raise ArgumentError("Invalid advertising version specified in args", supported=(1, 2), found=advertising_version)
```

With this method you don't need to save json_args around and people can look in a single place (this `__init__` function to see what you expect to find in args.

Also, AdvertisingPacketVer should conform to normal python naming convention of snake case since it's a normal class instance variable.